api/client/authn.go

```diff
@@ -0,0 +1,108 @@
+package client
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/pkg/term"
+)
+
+// ValidateAuthnOpt checks if a passed-in option value is a recognized
+// authentication option.
+func ValidateAuthnOpt(option string) (string, error) {
+	if strings.HasPrefix(option, "basic.username=") ||
+		strings.HasPrefix(option, "basic.password=") ||
+		strings.HasPrefix(option, "interactive=") ||
+		strings.HasPrefix(option, "bearer.token=") {
+		return option, nil
+	}
+	return "", fmt.Errorf("invalid authentication option %s", option)
+}
+
+func (cli *DockerCli) getBasic(challenge string, defusername string) (string, string, error) {
```

Comment:
Trying to pass in a default user name no longer made sense after the last rebase, so I'll replace this with just the realm name.  Thanks!